hls4ml/backends/fpga/fpga_backend.py

```diff
@@ -461,134 +475,135 @@ def compute_conv2d_instructions(self, in_H, in_W, in_C, kernel_size=3, stride=1,
-            return (
-                9,
-                9,
-                map(
-                    str,
-                    [
-                        1,
-                        3,
-                        7,
-                        15,
-                        31,
-                        30,
-                        28,
-                        24,
-                        16,
-                        33,
-                        99,
-                        231,
-                        495,
-                        1023,
-                        990,
-                        924,
-                        792,
-                        528,
-                        1057,
-                        3171,
-                        7399,
-                        15855,
-                        32767,
-                        31710,
-                        29596,
-                        25368,
-                        16912,
-                        33825,
-                        101475,
-                        236775,
-                        507375,
-                        1048575,
-                        1014750,
-                        947100,
-                        811800,
-                        541200,
-                        1082401,
-                        3247203,
-                        7576807,
-                        16236015,
-                        33554431,
-                        32472030,
-                        30307228,
-                        25977624,
-                        17318416,
-                        1082400,
-                        3247200,
-                        7576800,
-                        16236000,
-                        33554400,
-                        32472000,
-                        30307200,
-                        25977600,
-                        17318400,
-                        1082368,
-                        3247104,
-                        7576576,
-                        16235520,
-                        33553408,
-                        32471040,
-                        30306304,
-                        25976832,
-                        17317888,
-                        1081344,
-                        3244032,
-                        7569408,
-                        16220160,
-                        33521664,
-                        32440320,
-                        30277632,
-                        25952256,
-                        17301504,
-                        1048576,
-                        3145728,
-                        7340032,
-                        15728640,
-                        32505856,
-                        31457280,
-                        29360128,
-                        25165824,
-                        16777216,
-                    ],
-                ),
-            )
+        if not unscaled_W and not unscaled_H:
```

Comment:
Why not? And why only in 2D?

Reply:
I think you are right that the checks can be modified. I added this to fix a problem, but I can rework the checks to be more flexible.

Reply:
I modified this. Take a look to see if it makes sense. The issue I had previously was that min_W and min_H sometimes weren't the hardcoded values because of the modification due to too small of an in_width or in_height.